localstack/utils/aws/aws_stack.py

```diff
@@ -149,11 +150,12 @@ def get_partition(region_name: str = None):
     return boto3.session.Session().get_partition_for_region(region_name)
 
 
+# TODO: Deprecate and remove this
 def get_local_region():
     global LOCAL_REGION
     if LOCAL_REGION is None:
         LOCAL_REGION = get_boto3_region() or ""
-    return config.DEFAULT_REGION or LOCAL_REGION
+    return AWS_REGION_US_EAST_1 or LOCAL_REGION
```

Comment:
This can never fall back to `LOCAL_REGION`, can it? What is this function used for, and why should it be removed?

Reply:
This is function is related to the `Environment` abstractions defined in this file. It is currently used in Kinesis provider and DynamoDB provider. Ongoing work is focusing on use of `RequestContext.region` instead of these helpers, hence it will be removed together with `Environment` stuff

Reply:
This change mimics the current behaviour where `config.DEFAULT_REGION` will always be set

https://github.com/localstack/localstack/pull/9375/files#diff-8ee48b2bcad465473db1a1b202781ca2c7c23db7bd1670e79d45668fb2bd718fL371-L374